Mf2/Parser.php

```diff
@@ -778,9 +835,14 @@ public function parseDT(\DOMElement $dt, &$dates = array()) {
 		 * if $dtValue is only a time and there are recently parsed dates,
 		 * form the full date-time using the most recently parsed dt- value
 		 */
-		if ((preg_match('/^\d{1,2}:\d{1,2}(Z?[+|-]\d{2}:?\d{2})?/', $dtValue) or preg_match('/^\d{1,2}[a|p]m/', $dtValue)) && !empty($dates)) {
+		if ((preg_match('/^\d{1,2}:\d{1,2}(Z?[+|-]\d{2}:?\d{2}?)?/', $dtValue) or preg_match('/^\d{1,2}[a|p]m/', $dtValue)) && !empty($dates)) {
```

Comment:
What's the reason for making the minutes part optional here? It looks like the change is adding the `?` to the minutes part of the timezone offset. Like the above, this should probably be changed to `(Z|[+-]\d{2}:?\d{2})`

I know this wasn't your change, but `[a|p]` and `[+|-]` is incorrect, since the regex is using `[]` you only need `[ap]` and `[+-]`.